lightning-invoice/src/utils.rs

```diff
@@ -200,6 +202,37 @@ where
 		invoice = invoice.amount_milli_satoshis(amt);
 	}
 
+	for route_hint in select_phantom_hints(amt_msat, phantom_route_hints, logger) {
+		invoice = invoice.private_route(route_hint.clone());
```

Comment:
I believe we can just let `route_hint` move here and don't need to clone.

```suggestion
		invoice = invoice.private_route(route_hint);
```